File: linked-list/remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head==NULL){
            return NULL;
        }
        ListNode* p1=head;
        
        while(p1->next!=NULL&& p1!=NULL)
        {
            if(p1->val==p1->next->val){
                p1->next=p1->next->next;
            }
            else{
                p1=p1->next;
            }
            
        }
        return head;
        

        
    }
};

File: linked-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        if head==None:
            return None
        p1=head
        while(p1!=None and p1.next!=None):
            if(p1.val==p1.next.val):
                p1.next=p1.next.next
            else:
                p1=p1.next
        return head
       
        